mikupad.html

```diff
@@ -2324,6 +2342,20 @@
 			case 'false:false:false:Escape':
 				cancel();
 				break;
+			case 'false:true:false:r':
+			case 'false:false:true:r':
+				undoAndPredict();
+				break;
+			case 'false:true:false:z':
+			case 'false:false:true:z':
+				undo();
+				break;
+			case 'false:true:true:Z':
+			case 'false:true:false:y':
+			case 'false:false:true:y':
+				redo();
+				break;

```

Comment:
I think undo and redo should only let preventDefault be called if they are actually executed, otherwise the normal text undo will always be blocked.

Reply:
Also, the undo/redo shouldn't be executed if the prediction is in progress.